src/Loop.php

```diff
@@ -0,0 +1,50 @@
+<?php
+
+namespace React\EventLoop;
+
+/**
+ * The `Loop` class exists as a convenient way to get the currently relevant loop
+ */
+final class Loop
+{
+    /**
+     * @var LoopInterface
+     */
+    private static $instance;
+
+
+    /**
+     * Returns the event loop.
+     * When no loop is set it will it will call the factory to create one.
+     *
+     * This method always returns an instance implementing `LoopInterface`,
+     * the actual event loop implementation is an implementation detail.
+     *
+     * This method is the preferred way to get the event loop and using
+     * Factory::create has been deprecated.
+     *
+     * @return LoopInterface
+     */
+    public static function get()
+    {
+        if (self::$instance instanceof LoopInterface) {
+            return self::$instance;
+        }
+
+        self::$instance = Factory::create();
+
+        return self::$instance;
+    }
+
+    /**
+     * Internal undocumented method, behavior might change or throw in the
+     * future. Use with cation and at your own risk.
+     *
+     * @internal
```

Comment:
It seems strange to me to make this method internal. Users can then no longer decide to use a specific event loop implementation without using internal APIs, because the factory is the only public API, e.g. if both uv and ev are installed, a user might still prefer to use ev.

Reply:
Until we are sure about what such behaviour should be of the `Loop::set` method it will be marked as internal, and undocumented. You can still use it, but at your own risk. We don't want to introduce a function without being absolutely sure it is the right thing to do. Because once we add it, we can't remove it until the next major version

Reply:
> Users can then no longer decide to use a specific event loop implementation without using internal APIs […]

Any event loop can still be instantiated as usual if you need to make an explicit choice, this PR does not change anything about this. For the >99% use cases only a single event loop implementation would be available and the default choice is still reasonable.

> It seems strange to me to make this method internal.

I agree that this may appear odd on first sight. Like @WyriHaximus pointed out, we're using this as a starting point to be able to gather more feedback. Expect a public `set()` public not too far in the future in a separate PR :shipit: